### PR TITLE
Add test to check documentation on the sidebar

### PR DESCRIPTION
**Pull request by matiaswilner**

### CODE REVIEW

File: lib/elixir_console_web/live/console_live/sidebar_component.html.leex [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <% end %>
     </ul>
   </div>
-  <div class="flex-1 flex flex-col justify-end p-2">
+  <div id="documentation-output" class="flex-1 flex flex-col justify-end p-2">
```
Comment (eldano) [inline lib/elixir_console_web/live/console_live/sidebar_component.html.leex:14]: Not sure if this name is appropriate as this section shows other things that are not documentation. I can't think of a better name though as this shows different things. We could however, add this id or a similar one in one of the spans below, where we display the contextual help, or maybe even wrap those three spans with a div with the id. What do you think?

File: test/elixir_console_web/features/console_test.exs [hunk 2]
```diff
@@ -41,4 +42,19 @@ defmodule ElixirConsoleWeb.ConsoleTest do
...
+    |> visit("/")
+    |> fill_in(@command_input, with: "String.length('elixir')")
+    |> send_keys([:enter])
+    |> find(@command_output, fn output ->
+      output
+      |> assert_text("> String.length('elixir')")
+      |> click(Query.css("#command0 > span"))
+    end)
+    |> find(@documentation_output, fn documentation_output ->
+      assert_text(documentation_output, "String.length/1")
```
[the remark this answers is not in the document]
Comment (matiaswilner) [reply test/elixir_console_web/features/console_test.exs:57]: That is something we thought yesterday but didn't find any solution. Do you have any?

File: lib/elixir_console_web/live/console_live/sidebar_component.html.leex (hunk 1 above)
Comment (matiaswilner) [reply lib/elixir_console_web/live/console_live/sidebar_component.html.leex:14]: I agree with you. If we go for the second solution (a div wrapping the three spans), is there a possibility that the UI gets changed?

Comment (eldano) [reply lib/elixir_console_web/live/console_live/sidebar_component.html.leex:14]: I think a div wrapping the three spans gives semantics to that section so I see it as a nice addition, even if we didn't need it for the tests. Also I could see each section being a separate component in the future so in that case we would also need an element that wraps the component.
The UI should not be affected by adding a wrapping div as all padding/margin/borders default to zero.

File: test/elixir_console_web/features/console_test.exs [hunk 3]
```diff
@@ -41,4 +42,24 @@ defmodule ElixirConsoleWeb.ConsoleTest do
...
+
+  feature "visitor can get official documentation of an elixir function", %{session: session} do
+    session
+    |> visit("/")
+    |> fill_in(@command_input, with: "String.length('elixir')")
+    |> send_keys([:enter])
+    |> find(@command_output, fn output ->
+      output
+      |> assert_text("> String.length('elixir')")
+      |> click(css("#command0 > span"))
```
Comment (grzuy) [inline test/elixir_console_web/features/console_test.exs:54]: Wondering whether we could make this line a tiny bit more readable/understandable, in the sense that's attempting to click on the function link to documentation.

Thoughts on doing something like 

```diff
diff --git a/lib/elixir_console_web/live/console_live/history_component.ex b/lib/elixir_console_web/live/console_live/history_component.ex
index 8e1205f..dc89241 100644
--- a/lib/elixir_console_web/live/console_live/history_component.ex
+++ b/lib/elixir_console_web/live/console_live/history_component.ex
@@ -45,7 +45,7 @@ defmodule ElixirConsoleWeb.ConsoleLive.HistoryComponent do
          docs: docs,
          link: link
        }) do
-    ~e{<span
+    ~e{<a href="#"
     phx-click="function_link_clicked"
     phx-value-func_name="<%= func_name %>"
     phx-value-header="<%= header %>"
@@ -53,7 +53,7 @@ defmodule ElixirConsoleWeb.ConsoleLive.HistoryComponent do
     phx-value-link="<%= link %>"
     phx-target="#commandOutput"
     class="<%= inline_help_class_for(function_or_operator) %>"
-    ><%= part %></span>}
+    ><%= part %></a>}
   end
 
   defp inline_help_class_for(:function), do: "text-green-400 cursor-pointer underline"
diff --git a/test/elixir_console_web/features/console_test.exs b/test/elixir_console_web/features/console_test.exs
index 2f83316..462025d 100644
--- a/test/elixir_console_web/features/console_test.exs
+++ b/test/elixir_console_web/features/console_test.exs
@@ -2,7 +2,7 @@ defmodule ElixirConsoleWeb.ConsoleTest do
   use ExUnit.Case, async: true
   use Wallaby.Feature
 
-  import Wallaby.Query, only: [css: 1]
+  import Wallaby.Query, only: [css: 1, link: 1]
 
   @command_input css("#commandInput")
   @command_output css("#commandOutput")
@@ -85,7 +85,7 @@ defmodule ElixirConsoleWeb.ConsoleTest do
     |> find(@command_output, fn output ->
       output
       |> assert_text("> String.length('elixir')")
-      |> click(css("#command0 > span"))
+      |> click(link("String.length"))
     end)
     |> find(@documentation_output, fn documentation_output ->
       assert_text(documentation_output, "String.length/1")
```

?

If we prefer keep it as a span, maybe we just add a code comment in the test case explaining what the click is intended to click on the UI for the sake of better "understandability"?

**Review by grzuy — APPROVED** [review]
Very niiice!

File: test/elixir_console_web/features/console_test.exs (hunk 3 above)
Comment (grzuy) [reply test/elixir_console_web/features/console_test.exs:54]: cc/ @eldano

**Comment by grzuy** [comment]
Note this one has conflicts with latest updates on `master.`.

File: test/elixir_console_web/features/console_test.exs (hunk 3 above)
Comment (eldano) [reply test/elixir_console_web/features/console_test.exs:54]: The only difference I see is that on hover the browser (at least Chrome) shows in the bottom corner the url you'd be redirected by following the link. If it's a span it doesn't show anything.

Another option is to find the span by one if its attributes which is a bit more clear where you're clicking, although the css is a bit uglier:
```elixir
...
|> click(css("[phx-value-func_name='String.length/1']"))
```

Comment (grzuy) [reply test/elixir_console_web/features/console_test.exs:54]: > The only difference I see is that on hover the browser (at least Chrome) shows in the bottom corner the url you'd be redirected by following the link. If it's a span it doesn't show anything.

Ah, right... don't like that he.

> 
> Another option is to find the span by one if its attributes which is a bit more clear where you're clicking, although the css is a bit uglier:
> 
> ```elixir
> ...
> |> click(css("[phx-value-func_name='String.length/1']"))
> ```

Yeah...

Not ideal given it's still a bit to coupled with the implementation like the `#command0 span`, but I think it's a good compromise for this particular situation.

I'd go with this last option for now :-) :+1: 

cc/ @matiaswilner.

Comment (grzuy) [reply test/elixir_console_web/features/console_test.exs]: Thanks for the update!

**Review by grzuy — APPROVED** [review]
Pending merge conflicts resolution

File: test/elixir_console_web/features/console_test.exs [hunk 4]
```diff
@@ -42,6 +43,26 @@ defmodule ElixirConsoleWeb.ConsoleTest do
...
+    |> find(@documentation_output, fn documentation_output ->
+      assert_text(documentation_output, "String.length/1")
+
+      assert_text(
+        documentation_output,
+        "Returns the number of Unicode graphemes in a UTF-8 string."
+      )
+    end)
+  end
+
```
[the remark this answers is not in the document]
Comment (eldano) [reply test/elixir_console_web/features/console_test.exs]: I use the vscode extension Trailing Spaces that helps seeing these trailing spaces by highlighting them. You can also configure it to automatically delete them on save

**Comment by grzuy** [comment]
Feature test coverage 4/4 :heavy_check_mark: :clap:

**Pull request merged**